src/win/tty.c

```diff
@@ -463,25 +488,9 @@ static void uv_tty_queue_read_line(uv_loop_t* loop, uv_tty_t* handle) {
   }
   assert(handle->tty.rd.read_line_buffer.base != NULL);
 
-  /* Duplicate the console handle, so if we want to cancel the read, we can */
-  /* just close this handle duplicate. */
-  if (handle->tty.rd.read_line_handle == NULL) {
-    HANDLE this_process = GetCurrentProcess();
-    r = DuplicateHandle(this_process,
-                        handle->handle,
-                        this_process,
-                        &handle->tty.rd.read_line_handle,
-                        0,
-                        0,
-                        DUPLICATE_SAME_ACCESS);
-    if (!r) {
-      handle->tty.rd.read_line_handle = NULL;
-      SET_REQ_ERROR(req, GetLastError());
-      uv_insert_pending_req(loop, (uv_req_t*)req);
-      goto out;
-    }
-  }
-
+  /* Reset flags */
+  uv_read_console_started = FALSE;
+  uv_read_console_trap = FALSE;
```

Comment:
I don't think it's necessary because no other thread can be accessing these at this point (we are starting a read and so there cannot be one in progress). It's also relying on the assumption that there is a memory barrier inside QueueUserWorkItem(). I'll add a comment. I could also add an explicit MemoryBarrier() call, or Enter/LeaveCriticalSection, but it will have a small performance penalty.